bCNC/lib/bpath.py

```diff
@@ -467,22 +467,45 @@ def order(self, P):
 
 	#----------------------------------------------------------------------
 	# Intersect a line with line
+	# some explanations would be needed. replace by _intersectLineLineNEW ??
 	#----------------------------------------------------------------------
 	def _intersectLineLine(self, other):
 		# check for intersection
+# 		return self._intersectLineLineNEW(other)
 		DD = -self.AB[0]*other.AB[1] + self.AB[1]*other.AB[0]
 		if abs(DD)<EPS2: return None,None
 
 		Dt = -(other.A[0]-self.A[0])*other.AB[1] + \
 		      (other.A[1]-self.A[1])*other.AB[0]
-		t = Dt/DD
-		P = self.AB*t + self.A
+		t = Dt/DD #is this sure ? we have sin(AC,CM) / sin(AB,CD) see _intersectLineLineNEW (not used)
+		P = self.AB*t + self.A # is this sure in all cases ?
 		if self.minx<=P[0]<=self.maxx and other.minx<=P[0]<=other.maxx and \
-		   self.miny<=P[1]<=self.maxy and other.miny<=P[1]<=other.maxy:
+		   self.miny<=P[1]<=self.maxy and other.miny<=P[1]<=other.maxy:#is this sure??? what does a<=x<=b return ? Should't we write a<=x and x<=b instead ?
 			return P,None
 		return None,None
 
 	#----------------------------------------------------------------------
+	# Intersect a line with line (not used yet..)
```

Comment:
You complain about `_intersectLineLine` not being explained enough, but you don't provide a single line of explanation on how ` _intersectLineLineNEW` works nor how it is different from the original method...

Reply:
you're right Tomas
_interectLineLineNEW was just written for me, but I wanted to keep it somewhere on my branch I can remove it